x-pack/plugins/security_solution/public/explore/network/components/embeddables/embedded_map.tsx

```diff
@@ -165,6 +158,18 @@ export const EmbeddedMapComponent = ({
     };
   }, [addError, availableDataViews, isFieldInIndexPattern]);
 
+  useEffect(() => {
+    const dataViews = kibanaDataViews.filter((dataView) =>
+      selectedPatterns.includes(dataView.title)
+    );
+    if (selectedPatterns.length > 0 && dataViews.length === 0) {
+      setIsIndexError(true);
+    }
+    if (!isEqual(availableDataViews, dataViews)) {
+      setAvailableDataViews(dataViews);
+    }
+  }, [availableDataViews, kibanaDataViews, selectedPatterns]);
```

Comment:
We are setting the `availableDataViews` state here, which is also in the dependencies of the same useEffect. It is a bit weird. Could we do this instead?
```suggestion
    setAvailableDataViews((availableDataViews) =>
      isEqual(availableDataViews, dataViews) ? availableDataViews : dataViews
    );
  }, [kibanaDataViews, selectedPatterns]);
```

I am also wondering if the `isEqual` is really needed now. I believe that having the _memoizeOne_ inside the `isFieldInIndexPattern` we don't even need the `availableDataViews` state.